### PR TITLE
Various documentation improvements (#52600)

### PR DESCRIPTION
**Pull request by Hixie**
This is a reland of https://github.com/flutter/engine/pull/52600 which was reverted in https://github.com/flutter/engine/pull/52607 due to a typo.

### CODE REVIEW

**Review by jonahwilliams — APPROVED** [review]
LGTM!

**Comment by zanderso** [comment]
From PR triage: @Hixie are you still working on this?

**Comment by Hixie** [comment]
yup, just didn't want to land this when i wasn't around :-)

File: lib/ui/geometry.dart [hunk 1]
```diff
@@ -1749,6 +1752,17 @@ class RRect {
 ///
 /// Used by [Canvas.drawAtlas]. This is a more efficient way to represent these
 /// simple transformations than a full matrix.
+///
+/// It is equivalent to a [Matrix4] built using the following parameters (where
+/// `rst` is the corresponding `RSTransform` instance):
+///
+/// ```dart
+/// // Matrix4's constructor arguments are in column-major order.
+/// Matrix4(rst.scos, rst.ssin, 0, 0,
```
Comment (zanderso) [inline lib/ui/geometry.dart:1761]: snippet analysis is sad that `rst` isn't initialized.

Comment (zanderso) [reply lib/ui/geometry.dart:1761]: https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8748370038705622465/+/u/run_test.dart_for_analyze_shard_and_subshard_None/stdout

**Comment by zanderso** [comment]
Reason for revert: Snippet analysis failure blocking the roll to the framework.

**Comment by zanderso** [comment]
The revert bot appears to be down. I'll kick off a manual revert.

**Pull request merged**